### PR TITLE
minetest.wrap_text: Possible fix.

### PR DESCRIPTION
**Pull request by Tmanyo**
I created an issue recently about how minetest.wrap_text doesn't work.  The api says it will return a string.  It returns a table.  In the table, lines of text can be longer than the specified limit.  This is a function that I created and have found to work with the testing I have done.  Addresses #6033

### CODE REVIEW

**Comment by nerzhul** [comment]
why not just fix the doc ?

**Comment by Tmanyo** [comment]
Check the comment from Wuzzy2 on the issue I linked to.  He found that the function totally didn't work correctly.  From what I understand the function is supposed to wrap the text to the next line so that it fits in the specified character limit.  If we just change the documentation, what good is the function?

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -735,4 +719,4 @@ function core.privs_to_string(privs, delim)
 end
 
 assert(core.string_to_privs("a,b").b == true)
-assert(core.privs_to_string({a=true,b=true}) == "a,b")
+assert(core.privs_to_string({a=true,b=true}) == "a,b")
```
Comment (bigfoot547) [inline builtin/common/misc_helpers.lua]: Not high priority, but please configure your IDE to add a newline at the end of the file.

Comment (Tmanyo) [inline builtin/common/misc_helpers.lua]: Thank you for making me aware of what is causing that.  I noticed that it was the exact same line but I didn't realize it was caused by no newline at the end.

File: builtin/common/misc_helpers.lua [hunk 2]
```diff
@@ -308,59 +308,43 @@ function core.formspec_escape(text)
...
-		nl_start,nl_stop = string_find(text, "\n", current_idx)
-
-		if nl_start ~= nil and (start == nil or nl_start < start) then
-			start = nl_start
-			stop = nl_stop
-			gotnewline = true
+function core.wrap_text(text, limit)
+	local new_text = ""
+	local x = 1
+	if text:len() > limit then
```
Comment (SmallJoker) [inline builtin/common/misc_helpers.lua]: Use 
```Lua
if text:len() <= limit then
	return text
end
```
as no work has to be done anyway and it decreases the whole block below by one indent level.

File: builtin/common/misc_helpers.lua [hunk 3]
```diff
@@ -308,59 +308,43 @@ function core.formspec_escape(text)
...
+				else
+					local last_space = (limit - space) + 1
+					new_text = new_text .. 
+					text:sub(1,last_space) .. "\n"
+					text = text:sub(last_space + 1, 
+					text:len())
+				end
+				if text:len() < limit then
+					new_text = new_text .. text
+					text = ""
```
Comment (SmallJoker) [inline builtin/common/misc_helpers.lua]: Replace this line with `break` + see comments below

File: builtin/common/misc_helpers.lua [hunk 4]
```diff
@@ -308,59 +308,43 @@ function core.formspec_escape(text)
...
+				end
+			end
+			if text == "" then
+				x = 0
```
Comment (SmallJoker) [inline builtin/common/misc_helpers.lua]: `break`. The entire variable `x` can be removed when using the condition `true` -> `while true do`

**Comment by SmallJoker** [comment]
First of all. This function *must* return a table. It is used in first line to display the mod descriptions in the text list. If a string is desired, `table.concat` may be used.
Here's a reworked version where new lines can be forced in the input text: https://pastebin.com/xFjccg4x

**Comment by Tmanyo** [comment]
If the function _must_ return a table then the API _must_ be changed to show that it returns a table and not a string.

**Comment by Wuzzy2** [comment]
> This function must return a table. It is used in first line to display the mod descriptions in the text list.

Really? This is the only reason why it must be a table? Why not fix the main menu as well?

IMO it really doesn't make much sense for this function to return a table, it just seems very odd. Is there any REAL rationale behind returning a table?

But if you totally insist on a table, I won't start a war over it. What's more important is that the text wrapping as such works at least, and that the docs are correct.

**Comment by Tmanyo** [comment]
> But if you totally insist on a table, I won't start a war over it. What's more important is that the text wrapping as such works at least, and that the docs are correct.

I will not change the function I have to return a table as it makes no sense.

**Comment by SmallJoker** [comment]
@Tmanyo It's not known how many mods rely on this function. However, as it doesn't work properly anyway, it won't be used in many places. If you want to return a string, then please also fix the main menu, as Wuzzy suggested.

**Comment by SmallJoker** [comment]
Replaced by #6407.

**Pull request closed without merging**